pkg/kubefed2/federate/enable.go

```diff
@@ -124,32 +131,56 @@ func NewCmdFederateEnable(cmdOut io.Writer, config util.FedConfig) *cobra.Comman
 
 // Complete ensures that options are valid and marshals them if necessary.
 func (j *enableType) Complete(args []string) error {
+	j.federateDirective = &FederateDirective{}
+	ft := j.federateDirective
+
+	ft.Spec.ComparisonField = defaultComparisonField
+	ft.Spec.PrimitiveGroup = defaultPrimitiveGroup
+	ft.Spec.PrimitiveVersion = defaultPrimitiveVersion
+
+	if j.output == "yaml" {
+		j.outputYAML = true
+	} else if len(j.output) > 0 {
+		return fmt.Errorf("Invalid value for --output: %s", j.output)
+	}
+
+	if len(j.filename) > 0 {
+		return DecodeYAMLFromFile(j.filename, ft)
+	}
+
 	if len(args) == 0 {
 		return errors.New("NAME is required")
 	}
-	j.targetName = args[0]
+	ft.Name = args[0]
 
-	if j.rawComparisonField == string(apicommon.ResourceVersionField) ||
-		j.rawComparisonField == string(apicommon.GenerationField) {
-		j.comparisonField = apicommon.VersionComparisonField(j.rawComparisonField)
-	} else {
-		return fmt.Errorf("--comparison-field must be %q or %q",
-			apicommon.ResourceVersionField, apicommon.GenerationField,
-		)
+	if len(j.rawComparisonField) > 0 {
```

Comment:
Why is this check necessary?

Reply:
Good question.  Removed.

Reply:
I recall now that an earlier iteration of this PR allowed arguments to override what was provided in a file.  Since this argument is defaulted, it would override what appeared in the file by default.  I have since switched to not allowing arguments to override a directive supplied by file.